lib/rspec/mocks/space.rb

```diff
@@ -164,6 +171,24 @@ def proxy_not_found_for(id, object)
                       end
       end
 
+      def superclass_proxy_not_found_for(id, object)
+        raise "superclass_proxy_not_found_for called with something that is not a class" unless Class === object
+        proxies[id] = class_proxy_with_callback_verification_strategy(object, NoCallbackInvocationStrategy.new)
+      end
+
+      def class_proxy_with_callback_verification_strategy(object, strategy)
+        if RSpec::Mocks.configuration.verify_partial_doubles?
+          VerifyingPartialClassDoubleProxy.new(
+            self,
+            object,
+            @expectation_ordering,
+            strategy
+          )
+        else
+          PartialClassDoubleProxy.new(self, object, @expectation_ordering)
+        end
+      end
```

Comment:
I think the setter method also setting up the state is a bit weird, but I think we could have a method:

```
def partial_class_double_proxy_class
  {
    true => VerifyingPartialClassDoubleProxy,
    false => PartialClassdoubleProxy
  }.fetch(RSpec::Mocks.configuration.verify_partial_doubles?)
end
```
which would do the trick. Thoughts?